pkg/crd/api/targetconfig/v1alpha1/targetconfig_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TargetConfigSpec defines the desired state of TargetConfig.
+type TargetConfigSpec struct {
+	ConfigStrict `json:",inline"`
+
+	//+optional
+	S3 S3Options `json:"s3,omitempty"`
+
+	// +optional
+	Webhook WebhookOptions `json:"webhook,omitempty"`
+
+	// +optional
+	Telegram TelegramOptions `json:"telegram,omitempty"`
+
+	// +optional
+	Slack SlackOptions `json:"slack,omitempty"`
+
+	// +optional
+	ElasticSearch ElasticsearchOptions `json:"elasticSearch,omitempty"`
+
+	// +optional
+	GCS GCSOptions `json:"gcs,omitempty"`
+
+	TargetType string `json:"targetType,omitempty"`
```

Comment:
I don't think this is needed, you could make S3 etc. to pointers and check which property is set. You can add a schema validation that only one of this fields can be set and not multiple ones.